Chap_API_Server.tex

```diff
@@ -3333,6 +3334,62 @@ \subsection{\code{pmix_server_tool_connection2_fn_t}}
 the connection is refused, the \ac{PMIx} server library must terminate the
 connection attempt.
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\subsection{\code{pmix_server_tool_connection_fn_t}}
+\declareapi{pmix_server_tool_connection_fn_t}
+
+%%%%
+\summary
+
+Register that a tool has connected to the server.
+
+%%%%
+\format
+
+\copySignature{pmix_server_tool_connection_fn_t}{2.0}{
+typedef pmix_status_t (*pmix_server_tool_connection_fn_t)( \\
+\hspace*{29\sigspace}pmix_info_t info[], size_t ninfo, \\
+\hspace*{29\sigspace}pmix_tool_connection_cbfunc_t cbfunc, \\
+\hspace*{29\sigspace}void *cbdata);
+}
+
+\begin{arglist}
+\argin{info}{Array of \refstruct{pmix_info_t} structures (array of handles)}
+\argin{ninfo}{Number of elements in the \refarg{info} array (integer)}
+\argin{cbfunc}{Callback function \refapi{pmix_tool_connection_cbfunc_t} (function reference)}
+\argin{cbdata}{Data to be passed to the callback function (memory reference)}
+\end{arglist}
+
+\reqattrstart
+\ac{PMIx} libraries are required to pass the following attributes in the \refarg{info} array:
+
+\pasteAttributeItem{PMIX_USERID}
+\pasteAttributeItem{PMIX_GRPID}
+\pasteAttributeItemBegin{PMIX_TOOL_NSPACE}This must be included only if the tool already has an assigned namespace.
+\pasteAttributeItemEnd{}
+\pasteAttributeItemBegin{PMIX_TOOL_RANK}This must be included only if the tool already has an assigned rank.
+\pasteAttributeItemEnd{}
+\pasteAttributeItem{PMIX_CREDENTIAL}
+
+\reqattrend
+
+
+\optattrstart
+The following attributes are optional for host environments that support this operation:
+
+\pasteAttributeItem{PMIX_FWD_STDOUT}
+\pasteAttributeItem{PMIX_FWD_STDERR}
+\pasteAttributeItem{PMIX_FWD_STDIN}
+\pasteAttributeItem{PMIX_VERSION_INFO}
+
+\optattrend
+
+%%%%
+\descr
+
+This function module entry has been \textbf{DEPRECATED} in favor of \refapi{pmix_server_tool_connection2_fn_t}
+
+
```

Comment:
This is a mop-up/cleanup from a prior PR that introduced connection2?

Reply:
Yes - we have the unfortunate problem of conflicts between all these updates, so the diff once we merge becomes...a bloody mess.